Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,256 @@
+
+        private dynamic migrationHandlerInstance;
+
+        private bool SetMigrationHandlerInstance(Type type)
+        {
+            if (type == null || !typeof(IMigrationHandler).IsAssignableFrom(type))
+            {
+                Debug.LogError($"{type.Name} is not a valid implementation of IMigrationHandler");
+                return false;
+            }
+
+            if (!migrationHandlerTypes.Contains(type))
+            {
+                Debug.LogError($"{type.Name} might not be a valid implementation of IMigrationHandler");
+                return false;
+            }
+
+            try
+            {
+                migrationHandlerInstance = Activator.CreateInstance(type) as IMigrationHandler;
+            }
+            catch (Exception)
+            {
+                Debug.LogError("Selected MigrationHandler implementation could not be Instanciated");
+                return false;
+            }
+            return true;
+        }
+
+        public MigrationTool()
+        {
+            RefreshAvailableTypes();
+        }
+
+        private void RefreshAvailableTypes()
+        {
+            var type = typeof(IMigrationHandler);
+            migrationHandlerTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(x => x.GetLoadableTypes())
+                .Where(x => type.IsAssignableFrom(x) && !x.IsInterface && !x.IsAbstract).ToList();
+        }
+
+        /// <summary>
+        /// Adds Object to the list of objects to be migrated. Return false if Object is not of a GameObject, a Prefab or a Scene asset.
+        /// </summary>
+        /// <param name="selectedObject">Unity object of type of a GameObject or SceneAsset</param>
+        public bool TryAddObjectForMigration(Object selectedObject)
+        {
+            if (!selectedObject)
+            {
+                Debug.LogError("Selection is Empty. Could not add for Migration.");
+                return false;
+            }
+
+            if (selectedObject is GameObject || selectedObject is SceneAsset)
+            {
+                if (!migrationObjects.Contains(selectedObject))
+                {
+                    migrationObjects.Add(selectedObject);
+                    Debug.Log($"{selectedObject.name} object added for Migration");
+                }
+                return true;
+            }
+            Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for Migration");
+            return false;
+        }
+
+        /// <summary>
+        /// Adds all prefabs and scene assets found on the Assets folder to the list of objects to be migrated
+        /// </summary>
+        public void AddProjectForMigration()
+        {
+            AddAllAssetsOfTypeForMigration(new Type[] { typeof(GameObject), typeof(SceneAsset) });
+        }
+
+        private void AddAllAssetsOfTypeForMigration(Type[] types)
+        {
+            if (types != null)
```

Comment:
Should this be `types == null`? At the moment does this function return if types is initialized?

Reply:
Fixing